contracts/GenArt721CoreV3_Engine.sol

```diff
@@ -0,0 +1,2137 @@
+     * Does not account for refund if user overpays for a token (minter should
+     * handle a refund of the difference, if appropriate).
+     * Some minters may have alternative methods of splitting payments, in
+     * which case they should implement their own payment splitting logic.
+     * @param _projectId Project ID to be queried.
+     * @param _price Sale price of token.
+     * @return renderProviderRevenue_ amount of revenue to be sent to the
+     * render provider
+     * @return renderProviderAddress_ address to send render provider revenue to
+     * @return platformProviderRevenue_ amount of revenue to be sent to the
+     * platform provider
+     * @return platformProviderAddress_ address to send platform provider revenue to
+     * @return artistRevenue_ amount of revenue to be sent to Artist
+     * @return artistAddress_ address to send Artist revenue to. Will be null
+     * if no revenue is due to artist (gas optimization).
+     * @return additionalPayeePrimaryRevenue_ amount of revenue to be sent to
+     * additional payee for primary sales
+     * @return additionalPayeePrimaryAddress_ address to send Artist's
+     * additional payee for primary sales revenue to. Will be null if no
+     * revenue is due to additional payee for primary sales (gas optimization).
+     * @dev this always returns four addresses and four revenues, but if the
+     * revenue is zero, the corresponding address will be address(0). It is up
+     * to the contract performing the revenue split to handle this
+     * appropriately.
+     */
+    function getPrimaryRevenueSplits(uint256 _projectId, uint256 _price)
+        external
+        view
+        returns (
+            uint256 renderProviderRevenue_,
+            address payable renderProviderAddress_,
+            uint256 platformProviderRevenue_,
+            address payable platformProviderAddress_,
+            uint256 artistRevenue_,
+            address payable artistAddress_,
+            uint256 additionalPayeePrimaryRevenue_,
+            address payable additionalPayeePrimaryAddress_
+        )
+    {
+        ProjectFinance storage projectFinance = projectIdToFinancials[
+            _projectId
+        ];
+        // calculate revenues
+        uint256 projectFunds;
+        renderProviderRevenue_ =
+            (_price * uint256(_renderProviderPrimarySalesPercentage)) /
+            ONE_HUNDRED;
+        unchecked {
+            // renderProviderRevenue_ percentage is always <=100, so guaranteed to never underflow
+            projectFunds = _price - renderProviderRevenue_;
+        }
+        platformProviderRevenue_ =
+            (_price * uint256(_platformProviderPrimarySalesPercentage)) /
+            ONE_HUNDRED;
+        unchecked {
+            // platformProviderRevenue_ percentage is always <=100, so guaranteed to never underflow
+            projectFunds = _price - platformProviderRevenue_;
+        }
```

Comment:
I think would be good to add come comments here, and potentially move this outside of an unchecked block. This is now a three-way split of a pie, any loss of precision caused by a very low `_price` variable is now less straightforward to understand. In the case of a low price, Solidity should be rounding down when losing precision (we should confirm that in tests by testing that the artist receives all revenues when a sale price of 1 is input into this function), meaning that this should still never underflow. All things considered, I would lean towards keeping this out of an unchecked block, or writing enough tests here to ensure we feel very confident that Solidity's rounding down is working as intended. We should also add some inline comments that indicate why rounding down will prevent underflow here.

Reply:
Great point; agree with your rationale here–will update to do so (leaving this comment unresolved until I do so).

Reply:
Updated to remove unchecked logic and to expand on documentation a bit–can continue to iterate on this as needed.